OSSM-8338: Update gateway content (#84361)
openshift/openshift-docs: modules/ossm-exposing-a-gateway-to-traffic-outside-the-cluster-using-openshift-routes.adoc

.Prerequisites

> * You have completed the procedure, 'Exposing a Service by using the {istio} Gateway and VirtualService resources'

Author: done

.Procedure

> . Ensure that the `Service` type is set to `ClusterIP`.

Author: done

.Prerequisites

* You have completed the procedure, _Exposing a Service by using the {istio} Gateway and VirtualService resources_.

.Procedure

. Ensure that the `Service` type is set to `ClusterIP` by running the following command:
+
[source,terminal]
----
$ oc patch service  -n  -p '{"spec": {"type": "ClusterIP"}}'
----

> . Create a YAML file named `httpbin-route.yaml` that creates the `Route` service.

Reviewer: ```suggestion
. Create a YAML file named `httpbin-route.yaml` that defines a `Route` for the `httpbin` service.
```

Author: done

    name: <gateway_name>
    weight: 100
  wildcardPolicy: None
----

. Verify that `httpbin` service can be accessed from outside the cluster through the ingress router. Ensure that you set the `INGRESS_HOST` variable appropriately for the environment that your cluster is running in. 
+
.. If the cluster runs on AWS, run the following command to set the `INGRESS_HOST` variable:
[source,terminal]
----
$ INGRESS_HOST=$(oc get service router-default -n openshift-ingress -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
----
+
.. If the cluster runs on GCP or Azure, run the following command to set the `INGRESS_HOST` variable:
[source,terminal]
----
$ INGRESS_HOST=$(oc get service router-default -n openshift-ingress -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
----
+ 
.. Send a `curl` request to the `httpbin` service using the host of the ingress router:
[source,terminal]
----
$ curl -s -I -H Host:httpbin.example.com http://$INGRESS_HOST/headers
----
+

> Verify that the response has the `HTTP/1.1 200 OK` status, which indicates that the request was successful.

Author: The formatting looks like it was addressed

    name: <gateway_name>
    weight: 100
  wildcardPolicy: None
----

. Verify that `httpbin` service can be accessed from outside the cluster through the ingress router. Ensure that you set the `INGRESS_HOST` variable appropriately for the environment that your cluster is running in. 

> .. If the cluster runs on AWS, run the following command to set the `INGRESS_HOST` variable:

Reviewer 2: Totally up to you but it might follow our guidelines more closely (and tell users the important action sooner) if you say to run a command at the end:
```suggestion
.. If the cluster runs on AWS,  set the `INGRESS_HOST` variable by running the following command:
```

If you do decide to go with that suggestion, please also apply it to the next substep

Author: done

+
[source,terminal]
----
$ INGRESS_HOST=$(oc get service router-default -n openshift-ingress -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
----

.. If the cluster runs on GCP or Azure, run the following command to set the `INGRESS_HOST` variable:
+
[source,terminal]
----
$ INGRESS_HOST=$(oc get service router-default -n openshift-ingress -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
----
 
.. Send a `curl` request to the `httpbin` service using the host of the ingress router:
+
[source,terminal]
----
$ curl -s -I -H Host:httpbin.example.com http://$INGRESS_HOST/headers
----
+

> Verify that the response has the `HTTP/1.1 200 OK` status, which indicates that the request was successful.

Reviewer 2: This is arguably another instruction for users to follow and should be its own step:
```suggestion
. Verify that the response has the `HTTP/1.1 200 OK` status, which indicates that the request was successful.
```

Author: done
